### DISCUSSION TITLE
proximity-sort  Could not install because could not find 'unix' in 'os'

### DISCUSSION THREAD

**Discussion by MurraySobol** [immunant/c2rust#836] [General]
I recently upgraded Rust to latest version. 
I then ran "Cargo install c2rust" and got this error:

```
Compiling indexmap v1.9.2
error[E0433]: failed to resolve: could not find unix in os
--> C:\Users\murra.cargo\registry\src\github.com-1ecc6299db9ec823\c2rust-build-paths-0.17.0\src\lib.rs:35:26
|
35 | use std::os::unix::ffi::OsStrExt;
| ^^^^ could not find unix in os

Compiling proc-macro-error v1.0.4
error[E0599]: no function or associated item named from_bytes found for struct OsStr in the current scope
--> C:\Users\murra.cargo\registry\src\github.com-1ecc6299db9ec823\c2rust-build-paths-0.17.0\src\lib.rs:37:20
|
37 | OsStr::from_bytes(path)
| ^^^^^^^^^^ function or associated item not found in OsStr

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try rustc --explain E0433.
error: could not compile c2rust-build-paths due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile c2rust v0.17.0, intermediate artifacts can be found at C:\Users\murra\AppData\Local\Temp\cargo-installNnT2jx
```

I contacted Jon Gjengset (jonhoo) who discovered and corrected errors in "proximity-sort" which is now at version 1.3.0.

I feel that there has been a regression in this module in the last rust update; could someone review this module and (possibly) incorporate the changes made by jonhoo.

Thanks
Murray

**Comment by kkysen** [comment]
Hi @MurraySobol!  Building `c2rust` on Windows isn't currently supported.  Was it previously working for you?  This specific error is due to #708, which would be quite simple to fix itself (feel free to open a PR), but not worth it very much as there are other, much more complex issues with building `c2rust` on Windows which I imagine you'll run into next.

**Comment by MurraySobol** [comment]
kkysen: lets open a PR for this issue, I don't see that capability on this page.
To recap, It WAS working BEFORE I applied the latest Rust update, so I feel that this a regression issue.
I understand that building/running c2rust on Windows is not a high priority issue at this time but I hoe that this will change in the future.
 This is the reported issue: Fix some if cfg!(unix) mis-usages #708
